app/services/process_feed_service.rb

```diff
@@ -16,274 +16,14 @@ def update_author(body, account)
   end
 
   def process_entries(xml, account)
-    xml.xpath('//xmlns:entry', xmlns: TagManager::XMLNS).reverse_each.map { |entry| ProcessEntry.new.call(entry, account) }.compact
+    xml.xpath('//xmlns:entry', xmlns: TagManager::XMLNS).reverse_each.map { |entry| process_entry(entry, account) }.compact
   end
 
-  class ProcessEntry
-    def call(xml, account)
-      @account = account
-      @xml     = xml
-
-      return if skip_unsupported_type?
-
-      case verb
-      when :post, :share
-        return create_status
-      when :delete
-        return delete_status
-      end
-    rescue ActiveRecord::RecordInvalid => e
-      Rails.logger.debug "Nothing was saved for #{id} because: #{e}"
-      nil
-    end
-
-    private
-
-    def create_status
-      if redis.exists("delete_upon_arrival:#{@account.id}:#{id}")
-        Rails.logger.debug "Delete for status #{id} was queued, ignoring"
-        return
-      end
-
-      status, just_created = nil
-
-      Rails.logger.debug "Creating remote status #{id}"
-
-      if verb == :share
-        original_status = shared_status_from_xml(@xml.at_xpath('.//activity:object', activity: TagManager::AS_XMLNS))
-        return nil if original_status.nil?
-      end
-
-      ApplicationRecord.transaction do
```

Comment:
@akihikodaki It seems like the transaction is gone completely. I think I am experiencing glitches due to it (incomplete statuses in cache).